src/Sylius/Bundle/CoreBundle/PaymentRequest/Payum/Factory/PayumTokenFactory.php

```diff
@@ -45,15 +48,21 @@ public function createNew(PaymentRequestInterface $paymentRequest): TokenInterfa
          * }|null $payload
          */
         $payload = $paymentRequest->getPayload();
-        Assert::notNull($payload, 'The request payload need to be not null');
+        if (null === $payload) {
+            throw new InvalidPaymentRequestPayloadException('Payload of the payment request cannot be null');
+        }
 
         $targetPath = $payload['target_path'] ?? null;
```

Comment:
I didn't read sourcecode for this part, but in 2024 I would except a DTO with a data mapper or something :grimacing:

Reply:
Yeah I discussed with the team, we will probably refactor this in different PR